app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  layout 'standard'
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(params.require(:comment).permit(:text, :user_id, :post_id))
+    @comment.user = current_user
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+      redirect_to user_post_path(id: @comment.post_id, user_id: @comment.user_id)
+    else
+      flash.now[:error] = 'Error: Comment could not be created!'
+      render :new, locals: { comment: @comment }
+    end
+  end
+end
```

Comment:
- [ ] (Optional) Consider adding the  strong parameters, here is a link to a resource: [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

Reply:
Thank you @Munsa1 !
Here I am using strong parameters on line 8, but I haven't encapsulated them. I'll do that definitely when I create #update method to keep my code DRY.